src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,277 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+//
+// This class is a port of the Mono managed implementation of the MD4 algorithm
+// and required to support NTLM in Android only.
+// It's an implementation detail and is not intended to be a public API.
+// Assuming that NTLM would be System.Net.Security, it makes sense to put MD4 here as well.
+//
+namespace System.Net.Security
+{
+    internal sealed class MD4
+    {
+        private byte[]? HashValue;
```

Comment:
A lot of this code is maintaining the accumulator nature of the HashAlgorithm base type.

Assuming it's only needed in a one-shot manner, I recommend discarding all of the instance methods and just do whatever work is required to have a non-allocating Span-based static method.

```C#
internal static void HashData(ReadOnlySpan<byte> source, Span<byte> destination)
{
    Debug.Assert(destination.Length == 128 >> 3);
    Span<byte> buffer = stackalloc byte[64];
    etc
}
```